src/com/sap/piper/Utils.groovy

```diff
@@ -65,12 +65,14 @@ def stashStageFiles(Script script, String stageName) {
 }
 
 def unstashStageFiles(Script script, String stageName, List stashContent = []) {
-    stashContent += script.commonPipelineEnvironment.configuration.stageStashes?.get(stageName)?.unstash ?: []
+    def stashes = [] as Set
+    stashes += stashContent
+    stashes += script.commonPipelineEnvironment.configuration.stageStashes?.get(stageName)?.unstash ?: []
 
     script.deleteDir()
-    unstashAll(stashContent)
+    unstashAll(stashes)
 
-    return stashContent
+    return stashes
```

Comment:
We now return a `Set` instead of a `List`.
Could this cause problems for the callers?

Reply:
PS: `unstashAll` seems to return only the unstashed (successfully?) stashes. Wouldn't this be the expected behaviour here as well?

PPS: I wonder if `unstashedContent += unstash(stashContent[i])` in [line 111](https://github.com/SAP/jenkins-library/pull/4377/files#diff-f8351446966fe1c7d84083a9afbc0a6bace4ab0191fb4998bd6a8fffd506386eR111)  works. Jenkins `unstash` returns null in my tests.

Reply:
> PPS: I wonder if `unstashedContent += unstash(stashContent[i])` in [line 111](https://github.com/SAP/jenkins-library/pull/4377/files#diff-f8351446966fe1c7d84083a9afbc0a6bace4ab0191fb4998bd6a8fffd506386eR111) works. Jenkins `unstash` returns null in my tests.

The answer to this is: line 111 does not call the unstash step. It calls the unstash method of the Utils class. The unstash step is called there via `steps.unstash`. The reason why we have `steps` here is: the method hides the step with the same name. In order to call the step we have to say `steps.unstash` here. `steps` is (on Jenkins) a `org.jenkinsci.plugins.workflow.cps.DSL´. In the tests this gets mocked [here](https://github.com/SAP/jenkins-library/blob/master/test/groovy/util/BasePiperTest.groovy)